Operator IT: PodLogManager.stopAndWait() times out due to blocking readLine()

## Problem

The `PodLogManager` used in operator integration tests has a race condition in its
shutdown logic that causes `ConditionTimeoutException` during test teardown (`@AfterAll`).

**Observed behavior:** 8 out of 31 controller integration tests fail with:
```
org.awaitility.core.ConditionTimeoutException: Condition with Lambda expression
in io.apicurio.registry.operator.it.PodLogManager was not fulfilled within 5 minutes.
  at ITBase.afterAll:442
```

The actual test assertions pass — only the cleanup phase fails.

**Environment:** OCP 4.16 and 4.20 clusters provisioned via EBC (IBM CSD infrastructure).

## Root Cause

In `PodLogManager.java`:

1. `startPodLog()` (line 38-71) spawns a thread that calls `reader.readLine()` in a loop,
   checking a `stop` flag after each line.

2. `stopAndWait()` (line 78-81) sets the stop flag, then awaits the map to become empty:
   ```java
   activePodLogMap.values().forEach(stop -> stop.set(true));
   await().until(activePodLogMap::isEmpty);
   ```

3. **The problem:** `readLine()` at line 49 is a blocking call. If the pod is still running
   but not producing log output, `readLine()` blocks indefinitely. The `stop` flag is only
   checked *after* `readLine()` returns, so the thread cannot exit until either:
   - The pod produces another log line
   - The pod is deleted (returns `null`)
   - The stream is closed externally (throws exception)

   During `@AfterAll`, pods are still running (namespace deletion happens *after*
   `podLogManager.stopAndWait()` at line 458), so `readLine()` blocks and the
   Awaitility condition times out.

## Suggested Fixes

### Option A: Close the LogWatch to unblock readLine()

Store the `LogWatch` reference and close it during `stopAndWait()`. Closing the stream
causes `readLine()` to return `null`, which breaks the loop naturally.

```java
private final Map<ResourceID, LogWatch> activeLogWatchMap = new ConcurrentHashMap<>();

public void startPodLog(ResourceID podID) {
    new Thread(() -> {
        try (LogWatch logWatch = k8sClient.pods()...) {
            activeLogWatchMap.put(podID, logWatch);
            // ... existing read loop ...
        } finally {
            activeLogWatchMap.remove(podID);
            activePodLogMap.remove(podID);
        }
    }).start();
}

public void stopAndWait() {
    activePodLogMap.values().forEach(stop -> stop.set(true));
    activeLogWatchMap.values().forEach(lw -> {
        try { lw.close(); } catch (Exception e) { /* ignore */ }
    });
    await().atMost(Duration.ofSeconds(30)).until(activePodLogMap::isEmpty);
}
```

### Option B: Use non-blocking read with timeout

Replace the blocking `readLine()` with a `reader.ready()` check so the stop flag
is polled periodically:

```java
while (!stop.get()) {
    if (reader.ready()) {
        var line = reader.readLine();
        if (line == null) break;
        // ... process line ...
    } else {
        Thread.sleep(100);
    }
}
```

Option A is preferred as it's cleaner and doesn't introduce polling overhead.

## Re-opening: fix in #8509 / #8515 does not resolve the issue

The `LogWatch.close()` approach merged in #8509 (cherry-picked to `3.2.x` in #8515) was
confirmed present in a CSD pipeline run on 2026-07-13 (cloned `3.2.x` at HEAD, which
includes the fix). Both `SmokeITTest` and `MultiNamespaceSmokeITTest` still failed with
the same `ConditionTimeoutException` in `afterAll`:

```
[ERROR] MultiNamespaceSmokeITTest>ITBase.afterAll:442 » ConditionTimeout
  Condition with Lambda expression in io.apicurio.registry.operator.it.PodLogManager
  was not fulfilled within 30 seconds.

[ERROR] SmokeITTest>ITBase.afterAll:442 » ConditionTimeout
  Condition with Lambda expression in io.apicurio.registry.operator.it.PodLogManager
  was not fulfilled within 30 seconds.
```

The log threads started but never ended — `START LOG` entries are present for both test
classes but no corresponding `END LOG` entries appear, confirming that `readLine()` remained
blocked even after `logWatch.close()` was called.

### Why `LogWatch.close()` doesn't work

The fabric8 `LogWatch` implementation uses a `PipedInputStream`/`PipedOutputStream` pair
internally. Calling `logWatch.close()` closes the **output** side of the pipe (the side
that receives data from the Kubernetes API), but `BufferedReader.readLine()` is blocked
on the **input** side. Closing the output side of a `PipedOutputStream` does not
immediately unblock a `PipedInputStream.read()` — it only causes the *next* read attempt
to detect the broken pipe. If `readLine()` is mid-read waiting for a newline character, it
stays blocked.

### Suggested fix: interrupt the reader thread

Store the `Thread` reference and interrupt it during `stopAndWait()`. This will cause
`readLine()` to throw an `IOException` (or `InterruptedIOException`), which breaks the
loop and allows the `finally` block to clean up:

```java
private final Map<ResourceID, Thread> activeThreadMap = new ConcurrentHashMap<>();

public void startPodLog(ResourceID podID) {
    Thread thread = new Thread(() -> {
        // ... existing read loop ...
    });
    activeThreadMap.put(podID, thread);
    thread.start();
}

public void stopAndWait() {
    activePodLogMap.values().forEach(stop -> stop.set(true));
    activeLogWatchMap.values().forEach(logWatch -> {
        try { logWatch.close(); } catch (Exception e) { /* ignore */ }
    });
    activeThreadMap.values().forEach(Thread::interrupt);
    await().atMost(Duration.ofSeconds(30)).until(activePodLogMap::isEmpty);
}
```

The `catch (Exception ex)` block in the read loop will handle the `InterruptedIOException`
from the interrupted `readLine()`, and the `finally` block will remove entries from all
maps.

Alternatively, the non-blocking polling approach (Option B from the original issue) avoids
the problem entirely by never blocking on `readLine()`:

```java
while (!stop.get()) {
    if (reader.ready()) {
        var line = reader.readLine();
        if (line == null) break;
        // ... process line ...
    } else {
        Thread.sleep(100);
    }
}
```

### Test results context

- **Environment:** OCP 4.20 cluster provisioned via EBC
- **Branch:** `3.2.x` at HEAD (includes #8515)
- **Affected tests:** `MultiNamespaceSmokeITTest` (2 run, 1 error), `SmokeITTest` (6 run, 1 error)
- **Passing tests:** `SmokeOLMITTest` (1 run, 0 errors) — this test likely had active log
  output at teardown time, so `readLine()` returned naturally
- All actual test *assertions* passed — only `afterAll` cleanup fails

--- a/operator/controller/src/test/java/io/apicurio/registry/operator/it/PodLogManager.java
+++ b/operator/controller/src/test/java/io/apicurio/registry/operator/it/PodLogManager.java
@@ -48,17 +48,21 @@ public class PodLogManager {
                 activePodLogMap.put(podID, stop);
                 var lastWriteAt = Instant.now();
                 while (!stop.get()) {
-                    var line = reader.readLine();
-                    if (line != null) {
-                        chunk.append(getNamespace(podID)).append("/").append(podID.getName()).append(" >>> ")
-                                .append(line).append("\n");
+                    if (reader.ready()) {
+                        var line = reader.readLine();
+                        if (line == null) {
+                            break;
+                        }
+                        chunk.append(getNamespace(podID)).append("/").append(podID.getName())
+                                .append(" >>> ").append(line).append("\n");
                         if (lastWriteAt.plus(Duration.ofSeconds(5)).isBefore(Instant.now())) {
-                            log.debug("LOG of pod {}/{}:\n{}", getNamespace(podID), podID.getName(), chunk);
+                            log.debug("LOG of pod {}/{}:\n{}", getNamespace(podID), podID.getName(),
+                                    chunk);
                             chunk.setLength(0);
                             lastWriteAt = Instant.now();
                         }
                     } else {
-                        stop.set(true);
+                        Thread.sleep(100);
                     }
                 }
             } catch (Exception ex) {

